### PR TITLE
make Cluster be able to open file in classpath resource dir

### PR DESCRIPTION
**Pull request by BourneW**
i found Cluster.open() method can not load the yaml file that stored in project‘s resource dir,
therefore i want to submit this code to make this happen for  those who stored yaml file in there classpath based dir

### CODE REVIEW

File: gremlin-driver/src/main/java/org/apache/tinkerpop/gremlin/driver/Cluster.java [hunk 1]
```diff
@@ -44,10 +44,7 @@
 import java.io.IOException;
 import java.io.InputStream;
 import java.lang.ref.WeakReference;
-import java.net.InetAddress;
-import java.net.InetSocketAddress;
-import java.net.URI;
-import java.net.UnknownHostException;
+import java.net.*;
```
Comment (spmallette) [inline gremlin-driver/src/main/java/org/apache/tinkerpop/gremlin/driver/Cluster.java]: it looks like your IDE wildcarded our imports. could you please revert that part of your change?

File: gremlin-driver/src/main/java/org/apache/tinkerpop/gremlin/driver/Cluster.java [hunk 2]
```diff
@@ -244,9 +241,15 @@ public static Cluster open(final Configuration conf) {
      * Create a {@code Cluster} using a YAML-based configuration file.
      */
     public static Cluster open(final String configurationFile) throws Exception {
-        final File file = new File(configurationFile);
-        if (!file.exists())
-            throw new IllegalArgumentException(String.format("Configuration file at %s does not exist", configurationFile));
+        File file = new File(configurationFile);
+        if (!file.exists()) {
+            ClassLoader currentClassLoader = Thread.currentThread().getContextClassLoader();
```
Comment (spmallette) [inline gremlin-driver/src/main/java/org/apache/tinkerpop/gremlin/driver/Cluster.java]: Note that our code style is to `final` variables where possible...could you please add those where necessary?

File: gremlin-driver/src/main/java/org/apache/tinkerpop/gremlin/driver/Cluster.java [hunk 3]
```diff
@@ -244,9 +241,15 @@ public static Cluster open(final Configuration conf) {
      * Create a {@code Cluster} using a YAML-based configuration file.
```
Comment (spmallette) [inline gremlin-driver/src/main/java/org/apache/tinkerpop/gremlin/driver/Cluster.java]: Could you please expand the javadoc here to explain that the method will first try to read the file from the file system and then from resources?

**Comment by spmallette** [comment]
Thanks for you contribution. A few points:

1. You have targeted the master branch with this change which is 3.5.0 which has no fixed release date at this time. I think you should re-target this PR to tp33 so that it can be applied from 3.3.x forward.
2. Please add a test for this change - perhaps just add a new test here: https://github.com/apache/tinkerpop/blob/tp33/gremlin-server/src/test/java/org/apache/tinkerpop/gremlin/server/GremlinDriverIntegrateTest.java
3. Please add a CHANGELOG entry that describes your change (though I don't mind doing that myself on merge).

File: gremlin-driver/src/main/java/org/apache/tinkerpop/gremlin/driver/Cluster.java (hunk 3 above)
Comment (BourneW) [reply gremlin-driver/src/main/java/org/apache/tinkerpop/gremlin/driver/Cluster.java]: thanks for the reply! i will do the modification later for the points you given

Comment (BourneW) [reply gremlin-driver/src/main/java/org/apache/tinkerpop/gremlin/driver/Cluster.java]: Could you please check the commit again. Wether it is ok to approve?

**Comment by spmallette** [comment]
I think you need to rebase this PR or something - please have a look at the commit history. It is dragging in a lot of extra commits.

**Comment by BourneW** [comment]
> I think you need to rebase this PR or something - please have a look at the commit history. It is dragging in a lot of extra commits.

I have rebase it onto branch tp33 and sort out my commits.

**Comment by spmallette** [comment]
VOTE +1

**Comment by BourneW** [comment]
> VOTE +1

I don't know how to add a CHANGE LOG entry, cloud you please doing that on merge(if the pr will being approved)

**Comment by spmallette** [comment]
i can do it on merge but basically you just add a bullet point to this file describing your change:

https://github.com/apache/tinkerpop/blob/tp33/CHANGELOG.asciidoc#tinkerpop-3310-release-date-not-officially-released-yet

**Comment by BourneW** [comment]
> i can do it on merge but basically you just add a bullet point to this file describing your change:
> 
> https://github.com/apache/tinkerpop/blob/tp33/CHANGELOG.asciidoc#tinkerpop-3310-release-date-not-officially-released-yet

Got it ,and I have update the CHANGELOG.

**Pull request merged**